File: system-test/dotnet/Driver.Adapter/Shop/Ui/Client/Pages/OrderDetailsPage.cs
using Driver.Adapter.Shared.Client.Playwright;

using Driver.Port.Shop.Dtos;



namespace Driver.Adapter.Shop.Ui.Client.Pages;



public class OrderDetailsPage : BasePage

{

    private const string OrderNumberOutputSelector = "[aria-label='Display Order Number']";

    private const string OrderTimestampOutputSelector = "[aria-label='Display Order Timestamp']";

    private const string SkuOutputSelector = "[aria-label='Display SKU']";

    private const string QuantityOutputSelector = "[aria-label='Display Quantity']";

    private const string UnitPriceOutputSelector = "[aria-label='Display Unit Price']";

    private const string TotalPriceOutputSelector = "[aria-label='Display Total Price']";

    private const string StatusOutputSelector = "[aria-label='Display Status']";



    // Display text constants

    private const string DollarSymbol = "$";



    // Enum parsing constants

    private const bool IgnoreCase = true;



    public OrderDetailsPage(PageClient pageClient) : base(pageClient)

    {

    }



    public async Task<bool> IsLoadedSuccessfullyAsync()

    {

        return await PageClient.IsVisibleAsync(OrderNumberOutputSelector);

    }



    public async Task<string> GetOrderNumberAsync()

    {

        return await PageClient.ReadTextContentAsync(OrderNumberOutputSelector);

    }



    public async Task<DateTimeOffset> GetOrderTimestampAsync()

    {

        var textContent = await PageClient.ReadTextContentAsync(OrderTimestampOutputSelector);

        return DateTimeOffset.Parse(textContent, System.Globalization.CultureInfo.InvariantCulture);

    }



    public async Task<string> GetSkuAsync()

    {

        return await PageClient.ReadTextContentAsync(SkuOutputSelector);

    }



    public async Task<int> GetQuantityAsync()

    {

        var textContent = await PageClient.ReadTextContentAsync(QuantityOutputSelector);

        return int.Parse(textContent);

    }



    public async Task<decimal> GetUnitPriceAsync()

    {

        return await ReadTextMoneyAsync(UnitPriceOutputSelector);

    }



    public async Task<decimal> GetTotalPriceAsync()

    {

        return await ReadTextMoneyAsync(TotalPriceOutputSelector);

    }



    public async Task<OrderStatus> GetStatusAsync()

    {

        var status = await PageClient.ReadTextContentAsync(StatusOutputSelector);

        return Enum.Parse<OrderStatus>(status, IgnoreCase);

    }



    private async Task<decimal> ReadTextMoneyAsync(string selector)

    {

        var textContent = await PageClient.ReadTextContentAsync(selector);

        var cleaned = textContent.Replace(DollarSymbol, "").Trim();

        return decimal.Parse(cleaned);

    }



}

File: system-test/java/src/main/java/com/optivem/shop/dsl/driver/adapter/shop/ui/client/pages/OrderDetailsPage.java
package com.optivem.shop.dsl.driver.adapter.shop.ui.client.pages;

import com.optivem.shop.dsl.common.Converter;
import com.optivem.shop.dsl.driver.port.shop.dtos.OrderStatus;
import com.optivem.shop.dsl.driver.adapter.shared.client.playwright.PageClient;

import java.math.BigDecimal;
import java.time.Instant;


public class OrderDetailsPage extends BasePage {
    private static final String ORDER_NUMBER_OUTPUT_SELECTOR = "[aria-label='Display Order Number']";
    private static final String ORDER_TIMESTAMP_OUTPUT_SELECTOR = "[aria-label='Display Order Timestamp']";
    private static final String SKU_OUTPUT_SELECTOR = "[aria-label='Display SKU']";
    private static final String QUANTITY_OUTPUT_SELECTOR = "[aria-label='Display Quantity']";
    private static final String UNIT_PRICE_OUTPUT_SELECTOR = "[aria-label='Display Unit Price']";
    private static final String TOTAL_PRICE_OUTPUT_SELECTOR = "[aria-label='Display Total Price']";
    private static final String STATUS_OUTPUT_SELECTOR = "[aria-label='Display Status']";
    private static final String DOLLAR_SYMBOL = "$";

    public OrderDetailsPage(PageClient pageClient) {
        super(pageClient);
    }

    public boolean isLoadedSuccessfully() {
        return pageClient.isVisible(ORDER_NUMBER_OUTPUT_SELECTOR);
    }

    public String getOrderNumber() {
        return pageClient.readTextContent(ORDER_NUMBER_OUTPUT_SELECTOR);
    }

    public Instant getOrderTimestamp() {
        var textContent = pageClient.readTextContent(ORDER_TIMESTAMP_OUTPUT_SELECTOR);
        return Converter.toInstant(textContent);
    }

    public String getSku() {
        return pageClient.readTextContent(SKU_OUTPUT_SELECTOR);
    }

    public int getQuantity() {
        var textContent = pageClient.readTextContent(QUANTITY_OUTPUT_SELECTOR);
        return Integer.parseInt(textContent);
    }

    public BigDecimal getUnitPrice() {
        return readTextMoney(UNIT_PRICE_OUTPUT_SELECTOR);
    }

    public BigDecimal getTotalPrice() {
        return readTextMoney(TOTAL_PRICE_OUTPUT_SELECTOR);
    }

    public OrderStatus getStatus() {
        var status = pageClient.readTextContent(STATUS_OUTPUT_SELECTOR);
        return OrderStatus.valueOf(status);
    }

    private BigDecimal readTextMoney(String selector) {
        var textContent = pageClient.readTextContent(selector);
        var cleaned = textContent.replace(DOLLAR_SYMBOL, "").trim();
        return new BigDecimal(cleaned);
    }

}




